File: asn1runtime/src/main/scala/org/asn1gen/runtime/AsnUtf8String.scala
package org.asn1gen.runtime

class AsnUtf8String(value: String) extends AsnCharacterString(value) {
  override def _desc: meta.AsnUtf8String = meta.AsnUtf8String

  def copy(value: String = this.value) = new AsnUtf8String(value)

  override def equals(that: Any): Boolean = that match {
    case that: AsnUtf8String => this.value == that.value
    case _ => false
  }
}

object AsnUtf8String extends AsnUtf8String("") {
  def apply(value: String): AsnUtf8String = new AsnUtf8String(value)
}

File: asn1jrt/src/main/java/org/asn1gen/runtime/java/AsnUtf8String.java
package org.asn1gen.runtime.java;

public class AsnUtf8String implements AsnType {
  public static final AsnUtf8String EMPTY = new AsnUtf8String("");
  
  public final String value;
  
  public AsnUtf8String(final String value) {
    this.value = value;
  }
}
